imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -710,27 +718,45 @@ def swapi_l1(file_path: str, data_version: str) -> xr.Dataset:
     xtce_definition = (
         f"{imap_module_directory}/swapi/packet_definitions/swapi_packet_definition.xml"
     )
-    datasets = packet_file_to_datasets(
-        file_path, xtce_definition, use_derived_value=False
-    )
+    l0_unpacked_dict = {}
+    l1_hk_ds = None
+    for file_path in dependencies:
+        if file_path.suffix == ".pkts":
+            l0_unpacked_dict = packet_file_to_datasets(
+                file_path, xtce_definition, use_derived_value=False
+            )
+        if file_path.suffix == ".cdf":
+            l1_hk_ds = load_cdf(file_path)
+
     processed_data = []
 
-    for apid, ds_data in datasets.items():
-        # Right now, we only process SWP_HK and SWP_SCI
-        # other packets are not process in this processing pipeline
-        # If appId is science, then the file should contain all data of science appId
+    # Right now, we only process SWP_HK and SWP_SCI.
+    # Other apId are not processed in this processing pipeline.
 
-        if apid == SWAPIAPID.SWP_SCI.value:
-            sci_dataset = process_swapi_science(
-                ds_data, datasets[SWAPIAPID.SWP_HK], data_version
-            )
-            processed_data.append(sci_dataset)
-        if apid == SWAPIAPID.SWP_HK.value:
-            # Add HK datalevel attrs
-            hk_attrs = ImapCdfAttributes()
-            hk_attrs.add_instrument_global_attrs("swapi")
-            hk_attrs.add_global_attribute("Data_version", data_version)
-            ds_data.attrs.update(hk_attrs.get_global_attributes("imap_swapi_l1_hk"))
-            processed_data.append(ds_data)
+    # Len of dependencies is 2 and l0_unpacked_dict[SWAPIAPID.SWP_HK] is not None
+    if len(dependencies) == 2 and l0_unpacked_dict[SWAPIAPID.SWP_SCI]:
```

Comment:
Above you are only reading in the l0_unpacked_dict in an if block, this will fail if that isn't present. Do we ever expect that to be missing here? The dictionary lookup will raise a KeyError if it isn't present too and not just skip this, so I think it should be written like the following.

```suggestion
    if len(dependencies) == 2 and l0_unpacked_dict.get(SWAPIAPID.SWP_SCI, None) is not None:
```